src/Board.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Created by Shreyansh97 on 7/9/19
+ */
+
+public class Board {
+  private int size;
+  private int[] board;
+  
+  public Board(int size) {
+    this.size = size;
+    board = new int[size + 1];
+    for (int i = 0; i <= size; i++) {
+      board[i] = i;
+    }
+  }
+  
+  private int find(int pos) {
+    if (board[pos] != pos) {
+      board[pos] = find(board[pos]);
+    }
+    return board[pos];
+  }
+  
+  public void addSnake(int head, int tail) {
+    assert head > tail;
+    board[head] = find(tail);
+  }
+  
+  public void addLadder(int start, int end) {
+    assert start < end;
+    board[start] = find(end);
```

Comment:
you will miss the cases where u came down from a snake and u can get ladder to go up.

Reply:
we made it an optional requirement

Reply:
@guptashailesh92 those cases should work. The find function will take care of those cases when moving using the getPositionAfterMove function

Reply:
@gcnit I know but it could be done easily so..